src/platform/plugins/shared/dashboard/public/dashboard_api/panels_manager.ts

```diff
@@ -126,6 +131,12 @@ export function initializePanelsManager(
       return children$.value[id] as ApiType;
     }
 
+    // if the panel is in a collapsed section and has never been built, then childApi will be undefined
+    const sectionId = panels$.value[id].gridData.sectionId;
+    if (sectionId && (sectionsManager.api.sections$.value ?? {})[sectionId]?.collapsed) {
```

Comment:
Instead of exposing internal sections details with `sectionsManager.api.sections$.value ?? {})[sectionId]?.collapsed`, how about just passing in a callback `isSectionCollapsed`?

Reply:
Added in 261abc06be31f41780832849529627662036d124 -  that is super clean 🤌